src/legacy/ui/public/validated_range/is_range_valid.js

```diff
@@ -37,22 +38,23 @@ export function isRangeValid(value, min, max) {
   let isValid = true;
   let errorMessage = '';
 
-  if ((!isLowerValueValid && isUpperValueValid) || (isLowerValueValid && !isUpperValueValid)) {
+  if (!allowEmptyRange && (!isLowerValueValid || !isUpperValueValid)) {
```

Comment:
these floatings `ifs` without an early break complicate the control-flow. Why not have them mutually exclusive? 

I made that change to a vanilla if-elseif (can revert back if there's another reason why we're having these override eachother)